source/extensions/filters/http/dynamic_forward_proxy/proxy_filter.cc

```diff
@@ -203,6 +202,15 @@ void ProxyFilter::onLoadDnsCacheComplete(
   ASSERT(circuit_breaker_ != nullptr);
   circuit_breaker_.reset();
 
+  if (!host_info->address()) {
+    // Generally in Envoy it is not Ok to send a local reply at 2 code points with the same
+    // details, but here we could leak prior queries if we differentiate new failures from
+    // cached failures, so intentionally reuse the details.
```

Comment:
Maybe we could chat offline, but I'm not sure that I really understand. Is this comment observing that in  ProxyFilter::decodeHeaders() we send the same details that we're sending here? And normally that would be verboten. But if we returned a new error code here, we'd leak to the caller that a previous resolution had failed? Is that a privacy leak or some such?

Reply:
yeah I was worried this is more confusing than it's worth. Looking at our TODOs I think I also want to add a response flag, then refactored, so I think I can just remove this comment.

largely when we sendLocalReply we're supposed to be able to use rc-details to track exactly what went wrong.  generally I'd want separate codes for synchronous-dns-fail and async-dns-fail but that leaks prior requests so I'm intentionally not doing it. I don't think anyone else would realize the quandry so removing for clarity :-P

Reply:
SGTM!